### DISCUSSION TITLE
Use of flag vs. role in Schematron

### DISCUSSION THREAD

**Discussion by klakegg** [OP-TED/eForms-SDK#539] [Q&A] [answered]
Hi,

I've noticed the following example in the schematron folder (newlines added for readability):

```xml
<assert
  id="BR-BT-00024-0207"
  diagnostics="BT-24-Lot"
  flag="LAWFULNESS" test="((not(fn:matches(normalize-space(cbc:Description[./@languageID = 'ENG']/normalize-space(text())), '[^a-zA-Z0-9_]sale[^a-zA-Z0-9_]'))) or not(cbc:Description/@languageID = 'ENG')) or not(cbc:Description)"
  >If BT-24-Lot is in English, it should not contain the word "sale"</assert>
```

If this `assert` results in a `failed-assert`, how are we supposed to handle this as there are no `@role` attribute for the `assert`?

**Comment by bertrand-lorentz** [comment, marked as answer]
Thanks for your feedback.
From what I can see, the assert you mention does have a "role" attribute:
https://github.com/OP-TED/eForms-SDK/blob/develop/schematrons/static/validation-stage-5.sch#L49

For "lawfulness" rule, the role is set to "WARN", as they don't cause a notice to be rejected.

From which version of the SDK did you get the assert ?
I also see that it has the text in English, and not the translation code like in the SDK.
